### PR TITLE
created all models and basic unit tests

### PR DESCRIPTION
**Pull request by judyzhang25**

### CODE REVIEW

File: backend/src/concept.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      type: Schema.Types.ObjectId, 
+      ref: 'Group', 
+      required: true
+    },
+    name: {
+      type: String,
+      required: true
+    },
+    media: { 
+      data: Buffer,
```
[the remark this answers is not in the document]
Comment (judyzhang25) [reply backend/src/concept.js]: Mongoose's Buffer datatype is equivalent to binary. The code snippet I followed online used Buffer's in order to store images. See concept_test.js to see an example and also the orig code snippet here: https://gist.github.com/aheckmann/2408370

File: backend/server.js [hunk 2]
```diff
@@ -32,58 +34,129 @@ app.use(bodyParser.urlencoded({ extended: false }));
...
-    });
-  }
-  data.message = message;
-  data.id = id;
-  data.save(err => {
+router.post("/putGroup", (req, res) => {
+  let group = new Group(req.body);
+
+  console.log(req.body);
+//  if ((!id && id !== 0) || !message) {
```
[the remark this answers is not in the document]
Comment (machang19) [reply backend/server.js]: putGroup is still there sorry I just have some dead code I will delete in the putGroup method

File: backend/server.js [hunk 3]
```diff
@@ -32,58 +34,129 @@ app.use(bodyParser.urlencoded({ extended: false }));
...
+router.get("/getGroup2", (req, res) => {
+   keyword = req.query.keyword;
+   Group.find({"keyword" : keyword}, (err, data) => {
     if (err) return res.json({ success: false, error: err });
     return res.json({ success: true, data: data });
   });
 });
 
 // this is our update method
 // this method overwrites existing data in our database
```
[the remark this answers is not in the document]
Comment (machang19) [reply backend/server.js]: Yeah I made all the group fields when we had the member model also. We can remove the update method if you don't want to have any unused endpoints

[the remark this answers is not in the document]
Comment (machang19) [reply backend/server.js]: Sorry group endpoints not fiels

**Pull request merged**